### PR TITLE
Userland: check null pointer, and add man page for the lspci program

### LINKED ISSUE #1247: Userland: Bug in the lspci program
@deoxxa @awesomekling In code below, `db` could be null, and causes UB.
https://github.com/SerenityOS/serenity/blob/ca4d4cac45eb3d5f6999eebd8cb0b358afce235c/Userland/lspci.cpp#L85
https://github.com/SerenityOS/serenity/blob/ca4d4cac45eb3d5f6999eebd8cb0b358afce235c/Userland/lspci.cpp#L90
https://github.com/SerenityOS/serenity/blob/ca4d4cac45eb3d5f6999eebd8cb0b358afce235c/Userland/lspci.cpp#L95

if opening  the PCI ID database failed, db would be null, as suggested in
https://github.com/SerenityOS/serenity/blob/ca4d4cac45eb3d5f6999eebd8cb0b358afce235c/Userland/lspci.cpp#L56-L58

### PR DESCRIPTION
**Pull request by howar6hill**
@deoxxa @awesomekling The `db` pointer could be null, and cause UB. Fixes #1247. Additionally, add man page for lspci.

### CODE REVIEW

**Comment by howar6hill** [comment]
@bugaevc Suggestions accepted. All changes are made.

File: Base/usr/share/man/man8/lspci.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Description
+
+lspci is a utility for displaying information about PCI buses in the system
+and devices connected to them. It shows a brief list of devices.
+
+## Files
+
+* `/proc/pci` - a list of pci devices.
+* `/res/pci.ids` - try to get corresponding vendor name, device name, class name from this database.
```
[the remark this answers is not in the document]
Comment (bugaevc) [reply Base/usr/share/man/man8/lspci.md]: I don't think you're getting this.

"`/proc/pci` - a list of connected pci devices, and information about them such as bus, slot, vendor_id, and device_id." is a fun fact about ProcFS, but it doesn't explain what it has to do with `lspci`, which is what this man page is about.

I would like to see something like "`/proc/pci` - source of the PCI devices list" or "`/proc/pci` - read by `lspi` to fetch the PCI devices list".

**Comment by bugaevc** [comment]
> Thanks for the suggestions. All changes are made.

Apparently not the suggestion about two comments. Let me even suggest proper commit messages for them:

1.
> Userland: Fix nullptr dereference if we fail to open the PCIDB
>
> In the code below, db could be null, and would cause UB.
> Instead of crashing, let's simply skip symbolicating names.
>
> Fixes https://github.com/SerenityOS/serenity/issues/1247

2.
> Base: Add a man page for lspci

**Comment by howar6hill** [comment]
@bugaevc Is it better to open 2 new PR?

**Comment by bugaevc** [comment]
No, it's better to push 2 commits to this one PR :smile:

**Comment by howar6hill** [comment]
@bugaevc I added 2 commits.

**Comment by bugaevc** [comment]
:+1:

**Pull request merged**